src/ml_tooling/result/viz/classification_viz.py

```diff
@@ -33,7 +36,15 @@ def confusion_matrix(self, normalized: bool = True, **kwargs) -> plt.Axes:
 
         with plt.style.context(self._config.STYLE_SHEET):
             title = f"Confusion Matrix - {self._estimator_name}"
-            y_pred = self._estimator.predict(self._data.test_x)
+            if threshold is None:
+                y_pred = self._estimator.predict(self._data.test_x)
+            else:
+                y_prob = self._estimator.predict_proba(self._data.test_x)
```

Comment:
I think this does the same - should maybe be `preds >= threshold`? Check sklearn implementation?

```
y_prob = self._estimator.predict_proba(self._data.test_x)
y_pred = np.where((preds > threshold) & (preds == preds.max(axis=1, keepdims=True)), 1, 0)
```

Reply:
Seem rights, ill have to make tests as you suggest though. But this certainly assumes that the 0 class is indeed the false class and 1 the true, which i dont know is always the case? That was the reason for the last line in my suggestion

Reply:
Yes, you're right - need to add the indexing part:

`self._estimator.classes_[np.argmax(y_pred, axis=1)]`